client/scripts/views/components/new_thread_form.ts

```diff
@@ -134,42 +143,120 @@ export const NewThreadForm: m.Component<{
 }, {
   activeTag: OffchainTag | string,
   autoTitleOverride,
-  error,
   form: IThreadForm,
   fromDraft?: number,
   newType: string,
   quillEditorState,
+  recentlySaved: number[],
   saving: boolean,
   uploadsInProgress: number,
 }> = {
-  view: (vnode) => {
+  oninit: (vnode_) => {
+    console.log('initing');
+    console.log(vnode_.state.form);
+    vnode_.state.form = {};
+    vnode_.state.recentlySaved = [];
+    vnode_.state.uploadsInProgress = 0;
+    if (vnode_.state.newType === undefined) {
+      vnode_.state.newType = localStorage.getItem(`${app.activeId()}-post-type`) || PostType.Discussion;
+    }
+    if (vnode_.state.newType === PostType.Discussion) {
+      vnode_.state.form.threadTitle = localStorage.getItem(`${app.activeId()}-new-discussion-storedTitle`);
+    } else {
+      vnode_.state.form.url = localStorage.getItem(`${app.activeId()}-new-link-storedLink`);
+      vnode_.state.form.linkTitle = localStorage.getItem(`${app.activeId()}-new-link-storedTitle`);
+    }
+  },
+  view: (vnode: m.VnodeDOM<{
+    header: boolean,
+    isModal: boolean
+  }, {
+    activeTag: OffchainTag | string,
+    autoTitleOverride,
+    form: IThreadForm,
+    fromDraft?: number,
+    newType: string,
+    quillEditorState,
+    recentlySaved: number[],
+    saving: boolean,
+    uploadsInProgress: number,
+  }>) => {
+    if (!app.community && !app.chain) return;
     const author = app.user.activeAccount;
     const activeEntityInfo = app.community ? app.community.meta : app.chain.meta.chain;
+    const { isModal } = vnode.attrs;
     if (vnode.state.quillEditorState?.container) vnode.state.quillEditorState.container.tabIndex = 8;
 
     // init
-    if (vnode.state.form === undefined) vnode.state.form = {};
-    if (vnode.state.error === undefined) vnode.state.error = {};
-    if (vnode.state.uploadsInProgress === undefined) vnode.state.uploadsInProgress = 0;
-    if (vnode.state.newType === undefined) vnode.state.newType = 'Discussion';
-    const { error } = vnode.state;
-
+    // if (!vnode.state.recentlySaved) vnode.state.recentlySaved = [];
+    // if (vnode.state.form === undefined) vnode.state.form = {};
+    // if (vnode.state.error === undefined) vnode.state.error = {};
+    // if (vnode.state.uploadsInProgress === undefined) vnode.state.uploadsInProgress = 0;
+    // if (vnode.state.newType === undefined) {
+    //   vnode.state.newType = localStorage.getItem(`${app.activeId()}-post-type`);
+    //   if (!vnode.state.newType) vnode.state.newType = PostType.Discussion;
+    // }
```

Comment:
why comment these out?

Reply:
There was init logic in view that proxied for oninit by checking for undefineds.

Reply:
Okay, can we remove this altogether then?